esphome/components/tm1637/tm1637.cpp

```diff
@@ -173,8 +175,14 @@ void TM1637Display::display() {
   this->send_byte_(TM1637_I2C_COMM2);
 
   // Write the data bytes
-  for (auto b : this->buffer_) {
-    this->send_byte_(b);
+  if (this->inverted_) {
+    for (auto i = this->length_ - 1; i >= 0; i--) {
```

Comment:
I prefer being explicit about the type here, as if `auto` deduces an unsigned type, the loop will never terminate. It works fine currently because of the usual arithmetic conversions, but that's fragile (e.g. if `1` gets changed to `1u` it breaks).
```suggestion
    for (uint8_t i = this->length_ - 1; i >= 0; i--) {
```

Reply:
Looks like we caused what we tried to avoid ;)
See https://github.com/esphome/esphome/pull/2929